### PR TITLE
Retry on 500 from 'descriptor.{create,delete}'.

### PR DESCRIPTION
**Pull request by tseaver**
Toward #2176.

### CODE REVIEW

File: system_tests/monitoring.py [hunk 1]
```diff
@@ -24,6 +25,7 @@
 from system_test_utils import unique_resource_id
 
 retry_404 = RetryErrors(NotFound)
+retry_404_500 = RetryErrors(NotFound, InternalServerError)
```
Comment (dhermes) [inline system_tests/monitoring.py]: @tseaver doesn't the argument need to be a tuple? (Sorry on my phone so peeking at source is difficult)

File: system_tests/monitoring.py [hunk 2]
```diff
@@ -177,7 +179,7 @@ def test_create_and_delete_metric_descriptor(self):
         )
 
         descriptor.create()
-        retry_404(descriptor.delete)()
+        retry_404_500(descriptor.delete)()
```
Comment (dhermes) [inline system_tests/monitoring.py]: Check the issue. Delete isn't the only problem IIRC

**Comment by tseaver** [comment]
@dhermes  I just updated the commit to wrap a `retry_500` around the `descriptor.create`.  PTAL

File: system_tests/monitoring.py (hunk 1 above)
Comment (tseaver) [inline system_tests/monitoring.py]: Oops, missed that comment:  I just made it a tuple.

**Comment by daspecster** [comment]
LGTM, although I don't get why it's ok to throw a 500.
Is there a way that we can remember to undo this once the upstream service starts becoming more reliable?

**Comment by tseaver** [comment]
@daspecster How will we ever know:  that is a "Schroedinger's Cat" problem.

**Pull request merged**